neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/EmbeddedNeo4jDialect.java

```diff
@@ -533,6 +537,24 @@ public void forEachTuple(ModelConsumer consumer, TupleTypeContext tupleTypeConte
 		consumer.consume( new EmbeddedNeo4jTuplesSupplier( queryNodes, tupleTypeContext, entityKeyMetadata ) );
 	}
 
+	@Override
+	public ClosableIterator<Tuple> callStoredProcedure(String storedProcedureName,
+			ProcedureQueryParameters queryParameters, TupleContext tupleContext) {
+		Map.Entry<String, Map> queryAndParams = buildProcedureQueryWithParams( storedProcedureName, queryParameters );
+		Result result = dataBase.execute( queryAndParams.getKey(), queryAndParams.getValue() );
+		return CollectionHelper.newClosableIterator( extractTuples( result ) );
+	}
+
+	private List<Tuple> extractTuples(Result result) {
+		List<Tuple> tuples = new ArrayList<>();
+		while ( result.hasNext() ) {
+			Tuple tuple = new Tuple();
+			result.next().forEach( (key, value) -> tuple.put( key, value ) );
+			tuples.add( tuple );
+		}
+		return tuples;
```

Comment:
Similar to the previous comment, check what we do for `executeBackendQuery`. You should be able to reuse the code we have to convert the result.

Reply:
Hello @DavideD  is it about EmbeddedNeo4jBackendQueryResultIterator? Because it processes result slightly different, for example it change the order of parameters somehow in tests

Reply:
I'm afraid I'm not following you. Can you give me an example?
I don't see any difference between what you are doing and EmbeddedNeo4jBackendQueryResultIterator#convert (or better, mapSnapshot(...))

Reply:
@DavideD  I think, the problem is related to random order of columns in Result class, org.neo4j.graphdb.result.next which is called in EmbeddedNeo4jBackendQueryResultIterator return Map<String, Object> where order of keys are not the same as order of columns returned embedded neo4j. Result class provide a method org.neo4j.graphdb.result.columns which return the columns in correct order, but EmbeddedNeo4jBackendQueryResultIterator doesn't take it into account.

Reply:
@DavideD  I think, the problem is related to random order of columns in Result class, org.neo4j.graphdb.result.next which is called in EmbeddedNeo4jBackendQueryResultIterator return Map<String, Object> where order of keys are not the same as order of columns returned embedded neo4j. Result class provide a method org.neo4j.graphdb.result.columns which return the columns in correct order, but EmbeddedNeo4jBackendQueryResultIterator doesn't take it into account.

Reply:
@DavideD  I think, the problem is related to random order of columns in Result class, org.neo4j.graphdb.result.next which is called in EmbeddedNeo4jBackendQueryResultIterator return Map<String, Object> where order of keys are not the same as order of columns returned embedded neo4j.

Reply:
Hello, the problem here is result.next() returns Map<String, Object> where keys are not in the same order as returned columns returned by neo4j embedded.